File: Sorting/bubbleSort/bubbleSort.c
#include<stdio.h>

void bubbleSort(int a[]);

int main(){
     int arr[] = {2,3,57,8,9};
     
     bubbleSort(arr);

    return 0; 
}




void bubbleSort(int a[]){
       int i,j,temp;

       for (i=0;i<5;i++){

           for (j=i+1;j<5;j++){

               if(a[j]<a[i]){
                   temp = a[i];
                   a[i] = a[j];
                   a[j] = temp;
               }
           }
       }

       printf("Print sorted arr\n");

       for (i=0;i<5;i++){
           printf("%d ",a[i]);
       }
}

File: Sorting/bubbleSort/BubbleSort.java
class BubbleSort{

    public static void printArr(int arr[]){
        for(int i = 0; i < arr.length; i++){
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String[] args) {
        int arr[] = {8,5,6,5,4,3,9};


        //bubble sort
        for(int i = 0;i<arr.length-1;i++) {
            for(int j = 0; j<arr.length-i-1;j++){
                if(arr[j]>arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
            }
        }

        printArr(arr);
    }
}


//BubbleSort Time Complexity is 0(n^2)